### PR TITLE
Include elapsed time info in Progress/Summary HTTP headers

### PR DESCRIPTION
**Pull request by traceon**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Other

Changelog entry:
- None

Detailed description:
- An ad-hoc inclusion of elapsed time info in Progress/Summary HTTP headers
- This was started as a more coherent implementation in #18689

### CODE REVIEW

**Comment by vdimir** [comment]
Is it ok that if we perform query like `select * from table` we get only time before sending first byte of data in `X-ClickHouse-Summary`, but not time of whole query? 
It's limitation of headers itself, all of them sent before data, so it's impossible to get time of whole query in headers.
Buy with aggregated queries like `select sum(x) from table` elapsed time would be reasonable.

File: src/IO/WriteBufferFromHTTPServerResponse.h [hunk 1]
```diff
@@ -70,6 +70,7 @@ class WriteBufferFromHTTPServerResponse final : public BufferWithOwnMemory<Write
     Progress accumulated_progress;
     size_t send_progress_interval_ms = 100;
     Stopwatch progress_watch;
+    Stopwatch total_watch;
```
Comment (vdimir) [inline src/IO/WriteBufferFromHTTPServerResponse.h:73]: Maybe it's better to put into `Context`, set [here](https://github.com/ClickHouse/ClickHouse/blob/94c733bac76c2a33b1d4a79cafdfcfdad6f1bf10/src/Server/HTTPHandler.cpp#L735) and propagate in callback. But we can loose piece of time interval in summary if callback wouldn't called in time.

Comment (traceon) [reply src/IO/WriteBufferFromHTTPServerResponse.h:73]: Yeah, that makes sense. Which callback are you referring to?

**Comment by traceon** [comment]
> we get only time before sending first byte of data in X-ClickHouse-Summary, but not time of whole query?

I guess, that was the initial reason behind implementing time calculations in `SourceWithProgress`, in order to get the most "pure" and adequate values.

File: src/IO/WriteBufferFromHTTPServerResponse.h (hunk 1 above)
Comment (traceon) [reply src/IO/WriteBufferFromHTTPServerResponse.h:73]: Meanwhile, it looks like the instance of `WriteBufferFromHTTPServerResponse` is created before the query execution start, so the stopwatch start time should be adequate...

**Comment by vdimir** [comment]
@traceon And why using `system.query_log` is not suitable?

**Comment by traceon** [comment]
@vdimir we've discussed this internally, and will probably research a possibility to propose a separate PR with a configurable support for `HTTP Trailer` first (metadata at the end of chunked messages). Then this PR will make more sense, hopefully.

**Pull request closed without merging**